crates/polars-python/src/interop/arrow/to_rust.rs

```diff
@@ -97,6 +97,38 @@ pub fn to_rust_df(py: Python, rb: &[Bound<PyAny>], schema: Bound<PyAny>) -> PyRe
     let ArrowDataType::Struct(fields) = field_to_rust_arrow(schema)?.dtype else {
         return Err(PyPolarsErr::Other("invalid top-level schema".into()).into());
     };
+
+    {
+        // Verify that field names are not duplicated. Arrow permits duplicate field names, we do not.
```

Comment:
We can do this simpler. We can change the `DataFrame` constructor we call. E.g. we can call `new` which will already deduplicate.

If the output width differs from the input len, we had duplicates.